platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/ProxyFilterChain.java

```diff
@@ -14,84 +14,65 @@
 package org.codice.ddf.platform.filter.delegate;
 
 import java.io.IOException;
-import java.util.Arrays;
 import java.util.Iterator;
 import java.util.LinkedList;
-import java.util.List;
 
 import javax.servlet.Filter;
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 
+import org.codice.ddf.platform.filter.SecurityFilter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Implementation of filter chain that allows the ability to add new filters to
+ * Implementation of filter chain that allows the ability to add new {@link SecurityFilter}s to
  * a chain.
- *
  */
-public class ProxyFilterChain implements FilterChain {
+class ProxyFilterChain implements FilterChain {
     private static final Logger LOGGER = LoggerFactory.getLogger(ProxyFilterChain.class);
 
-    private FilterChain filterChain;
+    private final FilterChain filterChain;
 
-    private LinkedList<Filter> filters;
+    private final LinkedList<Filter> filters;
 
     private Iterator<Filter> iterator;
 
     /**
      * Creates a new ProxyFilterChain with the specified filter chain included.
      *
-     * @param filterChain
-     *            The filter chain from the web container.
+     * @param filterChain The filter chain from the web container.
      */
     public ProxyFilterChain(FilterChain filterChain) {
         this.filterChain = filterChain;
-        filters = new LinkedList<Filter>();
+        filters = new LinkedList<>();
     }
 
     /**
-     * Adds a single filter to the start of the local filter chain.
+     * Adds a single {@link SecurityFilter} to the start of the local filter chain.
      *
-     * @param filter
-     *            The servlet filter to add.
+     * @param filter The servlet filter to add.
      */
-    public void addFilter(Filter filter) {
-        if (filter != null) {
-            addFilters(Arrays.asList(filter));
-        } else {
+    public void addSecurityFilter(SecurityFilter filter) {
+        if (filter == null) {
             throw new IllegalArgumentException("Cannot add null filter to chain.");
         }
-    }
 
-    /**
-     * Adds a list of filters to the start of the local filter chain but before
-     * the initialized chain.
-     *
-     * @param filters
-     *            The servlet filters to add.
-     */
-    public void addFilters(List<Filter> filters) {
         if (iterator != null) {
```

Comment:
✏️ I know it's outside the scope of this change, but there is no concurrent access to this instance of the `ProxyFilterChain` (and if there were, this would not be anywhere near a sufficient test). While it's not specifically improper to check that the iterator is null and no chain is in progress, it does imply some things that aren't exactly true when viewed out of context.

The only value in the test is ensuring that no caller attempts to reuse this instance. That does hold some value, but I would add an inline comment to that effect here and update the javadocs on the class to make that clear.